protostar/commands/test/environments/setup_case_execution_environment.py

```diff
@@ -0,0 +1,22 @@
+from protostar.commands.test.cheatcodes.reflect.cairo_struct import CairoStructHintLocal
+from protostar.commands.test.environments.setup_execution_environment import (
+    SetupExecutionEnvironment,
+    SetupCheatcodeFactory,
+)
+from protostar.commands.test.test_context import TestContextHintLocal
+
+
+class SetupCaseExecutionEnvironment(SetupExecutionEnvironment):
+    async def invoke(self, function_name: str):
+        self.set_cheatcodes(SetupCaseCheatcodeFactory(self.state))
```

Comment:
```suggestion
    async def invoke(self, function_name: str):
        self.set_cheatcode_factory(self._setup_case_cheatcode_factory)
```
- [ ] If you inject the factory, dependencies can be injected directly to that factory instead of going through the ExecutionEnvironment.

Reply:
Cheatcode factories do not have uniform constructor interfaces. Additionally, in `FuzzTestExecutionStrategy` I am building the factory with a `strategy selector` object which is local to the particular `invoke` call. (Though this part is probably refactorable when we'll move `given` cheatcode to setup).

But in general, I would love to move factory construction to a separate method.

Reply:
> Cheatcode factories do not have uniform constructor interfaces.

I don't see a problem with that.

> Additionally, in FuzzTestExecutionStrategy I am building the factory with a strategy selector object which is local to the particular invoke call. 

I'm confused about this part, might come back to it in the future.